### ISSUE TITLE
Could not find suitable distribution for Requirement.parse('tensorflow>=1.1.0')

### ISSUE THREAD

**Issue by Anikacyp** [faneshion/MatchZoo#27]
When I trying to run "python setup.py install", it gives me this error:
Could not find suitable distribution for Requirement.parse('tensorflow>=1.1.0')
Is there anyone can help on this?

**Comment by faneshion** [comment]
It seems that error occurs while installing tensorflow. You can install tensorflow as well as other required software through pip.

**Comment by githubbaixiaobai** [comment]
I have the same error. But I have tensorflow(1.4.1) in my system and I have run program in tensorflow successfully  before. So is there anyone who can help us? 
And the error is listed below.

Searching for tensorflow>=1.1.0
Reading https://pypi.python.org/simple/tensorflow/
No local packages or download links found for tensorflow>=1.1.0
error: Could not find suitable distribution for Requirement.parse('tensorflow>=1.1.0')

**Comment by uduse** [comment]
@githubbaixiaobai @Anikacyp my guess is that you have old version of pip.

try:
`
pip install --upgrade pip
pip install tensorflow>=1.1.0
`

**Comment by githubbaixiaobai** [comment]
Firsly, I use 'pip uninstall tensorflow-gpu'  to uninstall tensorflow.
And then upgrade pip with 'pip install -upgrade pip' and reinstall tensorflow with 'pip install tensorflow-gpu'. The version of tensorflow is 1.4.1.
At last I run the command 'pip setup.py install --user'. 
Unfortunately, I got the same error

**Comment by githubbaixiaobai** [comment]
@uduse I think I solve the problem by replace 'tensorflow >= 1.1.0' with 'ensorflow-gpu >= 1.1.0'. and then run 'pip setup.py install --user '. After these two commands. I got:

Using /usr/lib/python2.7/site-packages
Finished processing dependencies for MatchZoo==0.2.0

Thanks for your help

**Comment by uduse** [comment]
@githubbaixiaobai aha, another conflicting environment problem, try setting up a virtual environment next time :)

**Comment by bwanglzu** [comment]
This is interesting that actually, MatchZoo should depend on Keras instead of Tensorflow. Since Keras is already a higher abstraction of several Deep Learning library.

One solution could be like [this](https://github.com/tensorflow/tensorflow/issues/7166), we move dependencies from `install_requires` to `extras_require` in `setup.py`, and let users decide which version of tensorflow to use.

A better solution is, we completely detach Tensorflow from MatchZoo, and let user decide which Keras backend they would like to use (e.g. tensorflow, tensorflow-gpu, theano, or even CNTK).

I'd like to re-open this issue because according to the GitHub insights, this issue has been viewed for over 300 times since past weeks. How do you think @uduse @faneshion @yangliuy 

![screenshot](https://image.ibb.co/jZPWT8/1.png)
